packages/material-ui/src/useMediaQuery/useMediaQuery.js

```diff
@@ -41,27 +57,20 @@ function useMediaQuery(queryInput, options = {}) {
       return undefined;
     }
 
-    const queryLists = queries.map(query => window.matchMedia(query));
-    setMatches(prev => {
-      const next = queryLists.map(queryList => queryList.matches);
-      return deepEqual(prev, next) ? prev : next;
-    });
+    const queryList = window.matchMedia(query);
+    setMatch(queryList.matches);
 
     function handleMatchesChange() {
-      setMatches(queryLists.map(queryList => queryList.matches));
+      setMatch(queryList.matches);
     }
 
-    queryLists.forEach(queryList => {
-      queryList.addListener(handleMatchesChange);
-    });
+    queryList.addListener(handleMatchesChange);
     return () => {
-      queryLists.forEach(queryList => {
-        queryList.removeListener(handleMatchesChange);
-      });
+      queryList.removeListener(handleMatchesChange);
     };
-  }, queries); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [query]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
What's preventing use from following rules of hooks now?

Reply:
Removing it, no longer needed.